chirp/drivers/ict10.py

```diff
@@ -0,0 +1,619 @@
+                    setattr(_settings, setting, element.value)
+            except Exception:
+                LOG.debug(element.get_name())
+                raise
+
+    def _get_extra(self, mem):
+        ext = RadioSettingGroup("extra", "Extra")
+
+        rev = RadioSetting("rev", "Reverse duplex",
+                           RadioSettingValueBoolean(bool(mem.rev)))
+        rev.set_doc("Reverse duplex")
+        ext.append(rev)
+
+        tx_inhibit = RadioSetting("tx_inhibit", "Tx inhibit",
+                                  RadioSettingValueBoolean(
+                                    bool(mem.tx_inhibit)))
+        tx_inhibit.set_doc("Tx inhibit")
+        ext.append(tx_inhibit)
+
+        return ext
+
+    def _get_memory(self, number, extd_number):
+        bit = 1 << (number % 8)
+        byte = int(number / 8)
+
+        mem = chirp_common.Memory()
+        mem.number = number
+
+        _mem = self._memobj.memory[number]
+
+        if extd_number is None:
+            _usd = self._memobj.used[byte]
+            _skp = self._memobj.skips[byte]
+            _psk = self._memobj.pskips[byte]
+        else:
+            mem.extd_number = extd_number
+            mem.immutable = ["number", "extd_number"]
+            if (number > 205):
+                mem.immutable.append("skip")
+            _usd = self._memobj.used[byte] if (number <= 205) else None
+            _skp = self._memobj.skips[byte] if (number <= 205) else None
+            _psk = self._memobj.pskips[byte] if (number <= 205) else None
+
+        if _usd is not None and (_usd & bit):
+            mem.empty = True
+            return mem
+
+        mem.freq = _mem.freq
+        mem.offset = int(_mem.offset)
+        mem.name = str(_mem.name).rstrip()
+        mem.rtone = chirp_common.TONES[_mem.rtone]
+        mem.ctone = chirp_common.TONES[_mem.ctone]
+        mem.dtcs = chirp_common.DTCS_CODES[_mem.dtcs]
+        mem.tuning_step = TUNING_STEPS[_mem.tuning_step]
+        mem.mode = MODES[_mem.mode]
+        mem.duplex = DUPLEXES[_mem.duplex]
+        mem.dtcs_polarity = DTCS_POLARITY[_mem.dtcs_polarity]
+        mem.tmode = TMODES[_mem.tmode]
+        mem.power = POWER_LEVELS[_mem.power]
+
+        mem.extra = self._get_extra(_mem)
+
+        if _skp is not None:
+            mem.skip = (_psk & bit and "P") or (_skp & bit and "S") or ""
+
+        return mem
+
+    def get_memory(self, number):
+        if not self._mmap:
+            self.sync_in()
+
+        assert (self._mmap)
+
+        extd_number = None
+        if isinstance(number, str):
+            try:
+                extd_number = number
+                number = dict(SPECIAL)[number]
+            except KeyError:
+                raise errors.InvalidMemoryLocation(
+                    "Unknown channel %s" % number)
+
+        return self._get_memory(number, extd_number)
+
+    def _fill_memory(self, number):
+        _mem = self._memobj.memory[number]
+
+        assert (_mem)
```

Comment:
Same here - this will never not be true, and it also isn't run in the built versions.